src/cmux.c

```diff
@@ -845,13 +845,14 @@ static rt_err_t cmux_vcom_open(rt_device_t dev, rt_uint16_t oflag)
 {
     rt_err_t result = RT_EOK;
     struct cmux *object = RT_NULL;
+    struct cmux_vcoms *vcom = (struct cmux_vcoms *)dev;
```

Comment:
"dev" is a pointer that virtual device, at the same time, this pointer also is the pointer that cmux_vcom structure.
I think this modification is wonderful choice, to reserverd space  for virtual device 's user data.

Thank you for improving this code.